### ISSUE TITLE
[rllib] Error when importing tensorflow before ray

### ISSUE THREAD

**Issue by stefanbschneider** [ray-project/ray#8993]
### What is the problem?

When importing `tensorflow` before `ray`, I get an error: `AttributeError: 'list' object has no attribute 'keys'`.
This does not happen if I remove the `tensorflow` import or import it after all `ray` and `rllib` imports.

I don't understand why this happens, but I guess it should not happen or at least lead to a more understandable error. It's very hard to debug otherwise.

I came across this bug here: https://github.com/ray-project/ray/issues/8971

*Ray version and other system information (Python version, TensorFlow version, OS):*

- Ray 0.8.5
- Tensorflow 2.2.0
- Python 3.8.3
- OS: Ubuntu 20.04 on WSL (Win 10)

### Reproduction (REQUIRED)
Please provide a script that can be run to reproduce the issue. The script should have **no external library dependencies** (i.e., use fake or mock data / environments):

If we cannot run your script, we cannot fix your issue.

- [x] I have verified my script runs in a clean environment and reproduces the issue.
- [x] I have verified the issue also occurs with the [latest wheels](https://docs.ray.io/en/latest/installation.html).

```python
import tensorflow
import ray
import ray.rllib.agents.ppo as ppo
from ray.tune.logger import pretty_print


ray.init()

# ray config
config = ppo.DEFAULT_CONFIG.copy()
config["num_workers"] = 1

# train on cartpole
trainer = ppo.PPOTrainer(config=config, env="CartPole-v0")
result = trainer.train()
pretty_print(result)
```

This leads to the aforementioned error for me. If I move the tensorflow import, it doesn't.

**Comment by robertnishihara** [comment]
@stefanbschneider can you include the full traceback?

**Comment by stefanbschneider** [comment]
Sure:

```
C:\Users\Stefan\AppData\Local\Microsoft\WindowsApps\ubuntu2004.exe run "export PYTHONUNBUFFERED=1 && export PYTHONIOENCODING=UTF-8 && export \"PYTHONPATH=/mnt/c/Users/Stefan/git-repos/work/deep-rl-mobility-management:/mnt/c/Program Files/JetBrains/PyCharm 2018.3.3/plugins/python/helpers/pycharm_matplotlib_backend:/mnt/c/Program Files/JetBrains/PyCharm 2018.3.3/plugins/python/helpers/pycharm_display\" && export PYCHARM_HOSTED=1 && export PYCHARM_DISPLAY_PORT=63342 && cd /mnt/c/Users/Stefan/git-repos/work/deep-rl-mobility-management && /mnt/c/Users/Stefan/git-repos/work/deep-rl-mobility-management/venv-wsl/bin/python /mnt/c/Users/Stefan/git-repos/work/deep-rl-mobility-management/rllib-import-error-example.py"
2020-06-18 07:56:18,535	INFO resource_spec.py:204 -- Starting Ray with 3.08 GiB memory available for workers and up to 1.54 GiB for objects. You can adjust these settings with ray.init(memory=<bytes>, object_store_memory=<bytes>).
2020-06-18 07:56:19,970	INFO services.py:1168 -- View the Ray dashboard at localhost:8265
E0618 07:56:21.314815900      14 socket_utils_common_posix.cc:208] check for SO_REUSEPORT: {"created":"@1592459781.314769300","description":"Protocol not available","errno":92,"file":"external/com_github_grpc_grpc/src/core/lib/iomgr/socket_utils_common_posix.cc","file_line":185,"os_error":"Protocol not available","syscall":"getsockopt(SO_REUSEPORT)"}
E0618 07:56:21.314972000      14 socket_utils_common_posix.cc:313] setsockopt(TCP_USER_TIMEOUT) Protocol not available
2020-06-18 07:56:22,895	INFO trainer.py:578 -- Current log_level is WARN. For more information, set 'log_level': 'INFO' / 'DEBUG' or use the -v and -vv flags.
WARNING:tensorflow:From /mnt/c/Users/Stefan/git-repos/work/deep-rl-mobility-management/venv-wsl/lib/python3.8/site-packages/ray/rllib/models/tf/tf_action_dist.py:84: multinomial (from tensorflow.python.ops.random_ops) is deprecated and will be removed in a future version.
Instructions for updating:
Use `tf.random.categorical` instead.
2020-06-18 07:56:23,867	INFO trainable.py:217 -- Getting current IP.
WARNING:tensorflow:From /mnt/c/Users/Stefan/git-repos/work/deep-rl-mobility-management/venv-wsl/lib/python3.8/site-packages/ray/rllib/policy/tf_policy.py:722: Variable.load (from tensorflow.python.ops.variables) is deprecated and will be removed in a future version.
Instructions for updating:
Prefer Variable.assign which has equivalent behavior in 2.X.
(pid=53) E0618 07:56:30.276113800      53 socket_utils_common_posix.cc:208] check for SO_REUSEPORT: {"created":"@1592459790.276096300","description":"Protocol not available","errno":92,"file":"external/com_github_grpc_grpc/src/core/lib/iomgr/socket_utils_common_posix.cc","file_line":185,"os_error":"Protocol not available","syscall":"getsockopt(SO_REUSEPORT)"}
(pid=53) E0618 07:56:30.276308500      53 socket_utils_common_posix.cc:313] setsockopt(TCP_USER_TIMEOUT) Protocol not available
(pid=raylet) E0618 07:56:20.682487000      36 socket_utils_common_posix.cc:208] check for SO_REUSEPORT: {"created":"@1592459780.682450400","description":"Protocol not available","errno":92,"file":"external/com_github_grpc_grpc/src/core/lib/iomgr/socket_utils_common_posix.cc","file_line":185,"os_error":"Protocol not available","syscall":"getsockopt(SO_REUSEPORT)"}
(pid=raylet) E0618 07:56:20.682851800      36 socket_utils_common_posix.cc:313] setsockopt(TCP_USER_TIMEOUT) Protocol not available
(pid=raylet) E0618 07:56:20.734644100      36 socket_utils_common_posix.cc:313] setsockopt(TCP_USER_TIMEOUT) Protocol not available
(pid=51) E0618 07:56:31.986794100      51 socket_utils_common_posix.cc:208] check for SO_REUSEPORT: {"created":"@1592459791.986772900","description":"Protocol not available","errno":92,"file":"external/com_github_grpc_grpc/src/core/lib/iomgr/socket_utils_common_posix.cc","file_line":185,"os_error":"Protocol not available","syscall":"getsockopt(SO_REUSEPORT)"}
(pid=51) E0618 07:56:31.987018900      51 socket_utils_common_posix.cc:313] setsockopt(TCP_USER_TIMEOUT) Protocol not available
(pid=52) E0618 07:56:32.161715000      52 socket_utils_common_posix.cc:208] check for SO_REUSEPORT: {"created":"@1592459792.161692800","description":"Protocol not available","errno":92,"file":"external/com_github_grpc_grpc/src/core/lib/iomgr/socket_utils_common_posix.cc","file_line":185,"os_error":"Protocol not available","syscall":"getsockopt(SO_REUSEPORT)"}
(pid=52) E0618 07:56:32.161938000      52 socket_utils_common_posix.cc:313] setsockopt(TCP_USER_TIMEOUT) Protocol not available
(pid=54) E0618 07:56:32.175404600      54 socket_utils_common_posix.cc:208] check for SO_REUSEPORT: {"created":"@1592459792.175382600","description":"Protocol not available","errno":92,"file":"external/com_github_grpc_grpc/src/core/lib/iomgr/socket_utils_common_posix.cc","file_line":185,"os_error":"Protocol not available","syscall":"getsockopt(SO_REUSEPORT)"}
(pid=54) E0618 07:56:32.175637200      54 socket_utils_common_posix.cc:313] setsockopt(TCP_USER_TIMEOUT) Protocol not available
2020-06-18 07:56:48,218	ERROR worker.py:1029 -- Possible unhandled error from worker: ray::RolloutWorker.set_weights() (pid=53, ip=192.168.178.24)
  File "python/ray/_raylet.pyx", line 463, in ray._raylet.execute_task
  File "python/ray/_raylet.pyx", line 417, in ray._raylet.execute_task.function_executor
  File "/mnt/c/Users/Stefan/git-repos/work/deep-rl-mobility-management/venv-wsl/lib/python3.8/site-packages/ray/rllib/evaluation/rollout_worker.py", line 571, in set_weights
    self.policy_map[pid].set_weights(w)
  File "/mnt/c/Users/Stefan/git-repos/work/deep-rl-mobility-management/venv-wsl/lib/python3.8/site-packages/ray/rllib/policy/tf_policy.py", line 357, in set_weights
    return self._variables.set_weights(weights)
  File "/mnt/c/Users/Stefan/git-repos/work/deep-rl-mobility-management/venv-wsl/lib/python3.8/site-packages/ray/experimental/tf_utils.py", line 182, in set_weights
    self.assignment_nodes[name] for name in new_weights.keys()
AttributeError: 'list' object has no attribute 'keys'

Process finished with exit code 0
```
